kivy/uix/textinput.py

```diff
@@ -1700,52 +1686,45 @@ def on_touch_up(self, touch):
+            # scroll_distance in less than the scroll_timeout since touch_down
+            if not (
+                _scroll_timeout <= self.scroll_timeout
+                and (
+                    (self._scroll_distance_x >= self.scroll_distance)
+                    or (self._scroll_distance_y >= self.scroll_distance)
+                )
             ):
-                self._enable_scroll = False
-                self._cancel_update_selection(self._touch_down)
-
-            if self._enable_scroll:
-                self.cancel_long_touch_event()
-                max_scroll_y = max(0, self.minimum_height - self.height)
-                self.scroll_y = min(max(0, self.scroll_y + touch.dy),
-                                    max_scroll_y)
-                self._trigger_update_graphics()
-                self._position_handles()
-                return True
+                # Distance isn't enough (yet) to consider it as a scroll
+                if _scroll_timeout <= self.scroll_timeout:
+                    # Timeout is not reached, scroll is still enabled.
+                    return False
+                else:
+                    self._enable_scroll = False
+                    self._cancel_update_selection(self._touch_down)
+                    return False
+            # We have a scroll!
+            self._have_scrolled = True
 
-        else:
-            _scroll_timeout = touch.time_update - touch.time_start
-            self._scroll_distance_x += abs(touch.dx)
+        self.cancel_long_touch_event()
 
-            # works with the same logic as multiline above
-            if (
-                _scroll_timeout >= self.scroll_timeout / 1000
-                and self._scroll_distance_x <= self.scroll_distance
-            ):
-                self._enable_scroll = False
-                self._cancel_update_selection(self._touch_down)
-
-            if self._enable_scroll:
-                self.cancel_long_touch_event()
-                minimum_width = (self._get_row_width(0) + self.padding[0] +
-                                 self.padding[2])
-                max_scroll_x = max(0, minimum_width - self.width)
-                self.scroll_x = min(max(0, self.scroll_x - touch.dx),
-                                    max_scroll_x)
-                self._trigger_update_graphics()
-                self._position_handles()
-                return True
+        if self.multiline:
+            max_scroll_y = max(0, self.minimum_height - self.height)
+            self.scroll_y = min(max(0, self.scroll_y + touch.dy),
+                                max_scroll_y)
+        else:
+            minimum_width = (self._get_row_width(0) + self.padding[0] +
+                                self.padding[2])
+            max_scroll_x = max(0, minimum_width - self.width)
+            self.scroll_x = min(max(0, self.scroll_x - touch.dx),
+                                max_scroll_x)
```

Comment:
```suggestion
            self.scroll_y = min(
                max(0, self.scroll_y + touch.dy),
                max_scroll_y
            )
        else:
            minimum_width = (
                self._get_row_width(0) + self.padding[0] + self.padding[2]
            )
            max_scroll_x = max(0, minimum_width - self.width)
            self.scroll_x = min(
                max(0, self.scroll_x - touch.dx),
                max_scroll_x
            )
```

A suggestion to improve readability.

Reply:
Considering that we're already touching these lines, looks good to me :)